app/views/collections/show.html.slim

```diff
@@ -25,19 +25,19 @@
           td
             = ''
       - else
-        - @collection.tasks.each do |exercise|
+        - @collection.tasks.each do |task|
           tr
             td
-              - if can? :read, exercise
-                = link_to exercise.title, exercise_path(exercise), class: 'btn btn-default'
+              - if can? :read, task
+                = link_to task.title, task_path(task), class: 'btn btn-default'
               - else
-                = link_to exercise.title, exercise_path(exercise), class: 'btn btn-default disabled'
+                = link_to task.title, task_path(task), class: 'btn btn-default disabled'
             td
-              - if cannot? :read, exercise
+              - if cannot? :read, task
                 span.fa.fa-lock.lock
                 = ' '
-                = link_to t('collections.show.contribute'), contribute_exercise_path(exercise), method: :post, class: 'btn btn-default'
-              = link_to remove_exercise_collection_path(@collection, exercise: exercise), method: :patch, class: 'pull-right' do
+                /= link_to t('collections.show.contribute'), contribute_task_path(task), method: :post, class: 'btn btn-default'
```

Comment:
As the previous commenting will also affect this line, we can revert the `/` here.

```suggestion
                = link_to t('collections.show.contribute'), contribute_task_path(task), method: :post, class: 'btn btn-default'
```